test/hotspot/jtreg/compiler/floatingpoint/TestRoundFloatAll.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8321010
+ * @summary Test intrinsic for Math.round(float) in full 32 bits range
+ *
+ * @library /test/lib /
+ * @modules java.base/jdk.internal.math
+ * @run main/othervm -XX:-TieredCompilation -XX:CompileThresholdScaling=0.3 -XX:+PrintIdeal -XX:CompileCommand=compileonly,compiler.floatingpoint.TestRoundFloatAll::test* -XX:-UseSuperWord compiler.floatingpoint.TestRoundFloatAll
+ */
+
+ package compiler.floatingpoint;
+
+import jdk.internal.math.FloatConsts;
+import compiler.lib.ir_framework.TestFramework;
+
+public class TestRoundFloatAll {
+
+  public static void main(String args[]) {
+    test();
+  }
+
+  static int golden_round(float a) {
+    // below code is copied from java.base/share/classes/java/lang/Math.java
+    //  public static int round(float a) { ... }
+
+    int intBits = Float.floatToRawIntBits(a);
+    int biasedExp = (intBits & FloatConsts.EXP_BIT_MASK)
+            >> (FloatConsts.SIGNIFICAND_WIDTH - 1);
+    int shift = (FloatConsts.SIGNIFICAND_WIDTH - 2
+            + FloatConsts.EXP_BIAS) - biasedExp;
+    if ((shift & -32) == 0) { // shift >= 0 && shift < 32
+        // a is a finite number such that pow(2,-32) <= ulp(a) < 1
+        int r = ((intBits & FloatConsts.SIGNIF_BIT_MASK)
+                | (FloatConsts.SIGNIF_BIT_MASK + 1));
+        if (intBits < 0) {
+            r = -r;
+        }
+        // In the comments below each Java expression evaluates to the value
+        // the corresponding mathematical expression:
+        // (r) evaluates to a / ulp(a)
+        // (r >> shift) evaluates to floor(a * 2)
+        // ((r >> shift) + 1) evaluates to floor((a + 1/2) * 2)
+        // (((r >> shift) + 1) >> 1) evaluates to floor(a + 1/2)
+        return ((r >> shift) + 1) >> 1;
+    } else {
+        // a is either
+        // - a finite number with abs(a) < exp(2,FloatConsts.SIGNIFICAND_WIDTH-32) < 1/2
+        // - a finite number with ulp(a) >= 1 and hence a is a mathematical integer
+        // - an infinity or NaN
+        return (int) a;
+    }
+  }
```

Comment:
At first, I was worried about the indentation, then realized the original code had the strange indentation.
Would there be a way to put this method in a shared file, so that you do not need to paste it everywhere?